src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -221,15 +224,22 @@ namespace winrt::TerminalApp::implementation
             }
             return false;
         };
-        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        static FromJsonResult FromJson(const Json::Value& json)
         {
             // LOAD BEARING: Not using make_self here _will_ break you in the future!
             auto args = winrt::make_self<ResizePaneArgs>();
             if (auto directionString{ json[JsonKey(DirectionKey)] })
             {
                 args->_Direction = ParseDirection(directionString.asString());
             }
-            return *args;
+            if (args->_Direction == TerminalApp::Direction::None)
+            {
+                return { nullptr, { ::TerminalApp::SettingsLoadWarnings::MissingRequiredParameter } };

```

Comment:
Just curious, you can still return `{ *args, { ... } }` right?

Reply:
I mean yea you could - but then the (theoretically) malformed args would still be added to the keybinding. Returning null will prevent the key fro getting bound.